--- libavcodec/cuviddec.c	2025-03-03 00:17:09.000000000 +0000
+++ libavcodec/cuviddec1.c	2025-07-23 13:58:08.337409400 +0100
@@ -878,7 +878,7 @@ static av_cold int cuvid_decode_init(AVC
                                    &ctx->resize.width, &ctx->resize.height) != 2) {
         av_log(avctx, AV_LOG_ERROR, "Invalid resize expressions\n");
         ret = AVERROR(EINVAL);
-        goto error;
+        return ret;
     }
 
     if (ctx->crop_expr && sscanf(ctx->crop_expr, "%dx%dx%dx%d",
@@ -886,13 +886,13 @@ static av_cold int cuvid_decode_init(AVC
                                  &ctx->crop.left, &ctx->crop.right) != 4) {
         av_log(avctx, AV_LOG_ERROR, "Invalid cropping expressions\n");
         ret = AVERROR(EINVAL);
-        goto error;
+        return ret;
     }
 
     ret = cuvid_load_functions(&ctx->cvdl, avctx);
     if (ret < 0) {
         av_log(avctx, AV_LOG_ERROR, "Failed loading nvcuvid.\n");
-        goto error;
+        return ret;
     }
 
     // respect the deprecated "surfaces" option if non-default value is given by user;
